arangod/Aql/RegisterPlan.cpp

```diff
@@ -83,6 +83,24 @@ void RegisterPlanWalkerT<T>::after(T* en) {
     }
   };
 
+  auto const updateRegsToKeep = [this](T* en, VarSet const& varsUsedLater,
+                                       VarSet const& varsValid) {
+    auto const& varsSetHere = en->getVariablesSetHere();
+
+    // regsToKeep
+    regsToKeepStack.back().clear();
```

Comment:
is it guaranteed here that `regsToKeepStack` is non-empty? if yes, then a comment here would help.

Reply:
This is what we expect. I should add a assertion as well as a comment here.

Reply:
Vale